### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by GrahamCampbell**
TODO:

* [x] Update composer.json
* [x] Fix source code
  - [x] Fix `FactoryParameter` for PHP 7.1-8.0
  - [x] Fix `FactoryGenerator` for newer versions of the finder
  - [x] Corrected generation to be platform independent (always generate unix style line endings)
  - [x] Regenerate functions file
* [x] Update tests and CI
  - [x] Test on PHP 8
  - [x] ~Fix coverage reporting~ Remove broken coverage reporting

---

EDIT 1: I will use GitHub Actions to run the PHP 7+ builds, and Travis for PHP 5 and HHVM 3.18.

EDIT 2: This repo is actually not enabled on scrutinizer or coveralls, and coverage reporting has been broken for years, so I have removed it.

---

Related to https://github.com/mockery/mockery/pull/1072, https://github.com/sebastianbergmann/phpunit/issues/4325.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Note that GitHub actions didn't run because the config is only present in my fork, but it will run along side travis once this is merged. The associated (passing) actions run from my fork can be found here: https://github.com/GrahamCampbell/hamcrest-php/runs/838598808?check_suite_focus=true.

**Comment by GrahamCampbell** [comment]
@davedevelopment This is ready for review now. :)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          extensions: curl
+          tools: composer:v2
+          coverage: none
+
+      - name: Install PHP 7 dependencies
+        run: composer update --prefer-dist --no-interaction --no-progress
+        if: "matrix.php != '8.0'"
+
+      - name: Install PHP 8 dependencies
+        run: composer update --prefer-dist --no-interaction --no-progress --ignore-platform-reqs
```
Comment (Ayesh) [inline .github/workflows/tests.yml:34]:
```suggestion
        run: composer update --prefer-dist --no-interaction --no-progress --ignore-platform-req=php
```

Wouldn't it be appropriate to use this, given composer v2 is used?

Comment (GrahamCampbell) [reply .github/workflows/tests.yml:34]: No, this can't be used, because the dependencies don't work on PHP 8.

**Comment by davedevelopment** [comment]
Thank you again Graham :raised_hands:

**Comment by GrahamCampbell** [comment]
@davedevelopment No problem at all. :) Let me know if there's anything else outstanding that you'd like help with before you're happy to tag the next release (`2.0.1` I assume). 🚢

**Pull request merged**